tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -9012,9 +9012,10 @@ SpirvEmitter::processIntrinsicInterlockedMethod(const CallExpr *expr,
   const auto srcLoc = expr->getExprLoc();
   const auto baseType = dest->getType()->getCanonicalTypeUnqualified();
 
-  if (!baseType->isIntegerType()) {
-    emitError("can only perform atomic operations on scalar integer values",
-              dest->getLocStart());
+  if (!baseType->isIntegerType() && !baseType->isFloatingType()) {
```

Comment:
I did not add more testing that this error is emitted, because in general is should never be. It is the job of the parser to determine if the correct types are used for the atomic operations.

Reply:
Would it make sense to make this an llvm_unreachable?

Reply:
Good call. I'll do that.